dask/array/routines.py

```diff
@@ -160,6 +161,57 @@ def dot(a, b):
     return tensordot(a, b, axes=((a.ndim - 1,), (b.ndim - 2,)))
 
 
+def _inner_apply_along_axis(arr,
+                            func1d,
+                            func1d_axis,
+                            func1d_args,
+                            func1d_kwargs):
+    return np.apply_along_axis(
+        func1d, func1d_axis, arr, *func1d_args, **func1d_kwargs
+    )
+
+
+@wraps(np.apply_along_axis)
+def apply_along_axis(func1d, axis, arr, *args, **kwargs):
+    arr = asarray(arr)
+
+    # Validate and normalize axis.
+    arr.shape[axis]
+    axis = len(arr.shape[:axis])
+
+    # Test out some data with the function.
+    test_data = np.ones((1,), dtype=arr.dtype)
+    test_result = np.array(func1d(test_data, *args, **kwargs))
+
+    if (LooseVersion(np.__version__) < LooseVersion("1.13.0") and
+            (np.array(test_result.shape) > 1).sum() > 1):
+            raise ValueError(
+                "Only one non-trivial dimension allowed in result. "
+                "Need NumPy 1.13.0+ for this functionality."
+            )
+
+    # Rechunk so that func1d is applied over the full axis.
+    arr = arr.rechunk(
+        arr.chunks[:axis] + (arr.shape[axis:axis + 1],) + arr.chunks[axis + 1:]
+    )
+
+    # Map func1d over the data to get the result
+    # Adds other axes as needed.
+    result = arr.map_blocks(
+        _inner_apply_along_axis,
```

Comment:
Realized that this was showing up in the Dask graph as the operation name. It certainly still can be understood what is going on, but it would be better to say `apply_along_axis` instead. Have fixed this in PR ( https://github.com/dask/dask/pull/2704 ) by specifying the `token` to be `"apply_along_axis"`. This fixes the name in the graph, but still uses Dask's standard method of hashing arguments.